src/robotFactory.js

```diff
@@ -1,10 +1,90 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion,
+  ) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    (!this.coords.x) && (this.coords.x = 0);
+    (!this.coords.y) && (this.coords.y = 0);
```

Comment:
useless check, maybe you need to chech coords from args
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0
    }
```